src/hotspot/share/runtime/synchronizer.hpp

```diff
@@ -172,29 +123,30 @@ class ObjectSynchronizer : AllStatic {
   // JNI detach support
   static void release_monitors_owned_by_thread(JavaThread* current);
 
+  // Iterate owned ObjectMonitors where the owner `filter` returns true.
+  template <typename OwnerFilter>
+  static void owned_monitors_iterate_filtered(MonitorClosure* closure, OwnerFilter filter);
+
   // Iterate ObjectMonitors where the owner == thread; this does NOT include
-  // ObjectMonitors where owner is set to a stack lock address in thread:
-  //
-  // This version of monitors_iterate() works with the in-use monitor list.
-  static void monitors_iterate(MonitorClosure* m, JavaThread* thread);
-  // This version of monitors_iterate() works with the specified linked list.
-  static void monitors_iterate(MonitorClosure* closure,
-                               ObjectMonitorsHashtable::PtrList* list,
-                               JavaThread* thread);
+  // ObjectMonitors where owner is set to a stack lock address in thread.
+  static void owned_monitors_iterate(MonitorClosure* m, JavaThread* thread);
+
+  // Iterate owned ObjectMonitors.
+  static void owned_monitors_iterate(MonitorClosure* closure);
```

Comment:
owned by whom? current thread? Does that include stack-locked or not?

Just trying to understand how the two variants of `owned_monitors_iterate` relate.

Reply:
Owned by *any* thread in any way.

* `owned_monitors_iterate(MonitorClosure* closure)` - Visits all monitors with the owner set to anything that indicates that the monitor has an owner (`ObjectMonitor::has_owner()`).

* `owned_monitors_iterate(MonitorClosure* m, JavaThread* thread)` - Visits all monitors with the owner field set to the specified `thread`.

Maybe we could figure out more descriptive names for these.

Reply:
Updating the comment would be fine:
```
// Iterate ObjectMonitors owned by any thread.
```
Thanks

Reply:
Done